<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TopWindow</class>
 <widget class="QWidget" name="TopWindow">
  <property name="windowModality">
   <enum>Qt::WindowModality::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>910</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>910</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>QString Class</string>
  </property>
  <property name="windowOpacity">
   <double>1.000000000000000</double>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="TopWindow_layout">
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="top_vlayout" stretch="2,3">
     <property name="spacing">
      <number>6</number>
     </property>
     <property name="sizeConstraint">
      <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
     </property>
     <property name="leftMargin">
      <number>6</number>
     </property>
     <property name="topMargin">
      <number>12</number>
     </property>
     <property name="rightMargin">
      <number>6</number>
     </property>
     <property name="bottomMargin">
      <number>9</number>
     </property>
     <item>
      <layout class="QVBoxLayout" name="L2_vlayout01">
       <property name="spacing">
        <number>1</number>
       </property>
       <item>
        <widget class="QGroupBox" name="input_g">
         <property name="minimumSize">
          <size>
           <width>877</width>
           <height>108</height>
          </size>
         </property>
         <property name="title">
          <string>GroupBox</string>
         </property>
         <layout class="QGridLayout" name="L3_glayout01" columnstretch="1,3,2">
          <property name="leftMargin">
           <number>9</number>
          </property>
          <property name="topMargin">
           <number>4</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <property name="verticalSpacing">
           <number>10</number>
          </property>
          <item row="0" column="2">
           <widget class="QPushButton" name="strunicode_phbton">
            <property name="text">
             <string>Every Char Unicode value</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="inch_label">
            <property name="text">
             <string>Inpu ANY Char</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="char_linedit">
            <property name="text">
             <string/>
            </property>
            <property name="placeholderText">
             <string>Any Unicode Single Char</string>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="str_linedit">
            <property name="inputMask">
             <string/>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="placeholderText">
             <string>Any Unicode String</string>
            </property>
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="instr_label">
            <property name="text">
             <string>Input ANY String</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="charfeat_phbton">
            <property name="text">
             <string>Single Char features</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="test_g">
         <property name="minimumSize">
          <size>
           <width>877</width>
           <height>107</height>
          </size>
         </property>
         <property name="title">
          <string>QString type cast test</string>
         </property>
         <widget class="QWidget" name="layoutWidget">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>40</y>
            <width>851</width>
            <height>51</height>
           </rect>
          </property>
          <layout class="QHBoxLayout" name="L3_hlayout02">
           <property name="spacing">
            <number>40</number>
           </property>
           <property name="leftMargin">
            <number>9</number>
           </property>
           <property name="topMargin">
            <number>4</number>
           </property>
           <property name="rightMargin">
            <number>9</number>
           </property>
           <property name="bottomMargin">
            <number>4</number>
           </property>
           <item>
            <widget class="QPushButton" name="tolatin_phbton">
             <property name="text">
              <string>Convert to Latin1</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="toutf16_phbton">
             <property name="text">
              <string>Convert to UTF-16</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="cmpqch_phbton">
             <property name="text">
              <string>QChar Compare</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="clcinput_phbton">
             <property name="text">
              <string>Clear Input Text</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="L2_hlayout00" stretch="2,1">
       <property name="spacing">
        <number>2</number>
       </property>
       <item>
        <widget class="QGroupBox" name="display_g">
         <property name="minimumSize">
          <size>
           <width>583</width>
           <height>324</height>
          </size>
         </property>
         <property name="title">
          <string>Result Display</string>
         </property>
         <widget class="QTextBrowser" name="display_result">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>30</y>
            <width>561</width>
            <height>291</height>
           </rect>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="features_g">
         <property name="minimumSize">
          <size>
           <width>288</width>
           <height>324</height>
          </size>
         </property>
         <property name="title">
          <string>QChar Features Check</string>
         </property>
         <layout class="QVBoxLayout" name="L3_hlayout00">
          <item>
           <widget class="QCheckBox" name="isdigit">
            <property name="text">
             <string>IsDigit</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="isletter">
            <property name="text">
             <string>isLetter</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="isletter_number">
            <property name="text">
             <string>isLetterOrNumber</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="isupper">
            <property name="text">
             <string>isUpper</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="islower">
            <property name="text">
             <string>isLower</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="ismark">
            <property name="text">
             <string>isMark</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="isspace">
            <property name="text">
             <string>isSpace</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="issymbol">
            <property name="text">
             <string>isSymbol</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="ispunct">
            <property name="text">
             <string>isPunct</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
